strax/storage/mongo.py

```diff
@@ -11,100 +11,159 @@
 import strax
 import numpy as np
 from pymongo import MongoClient
-from strax import StorageFrontend, StorageBackend, DataKey
-from strax.utils import unpack_dtype
-from bson import json_util
-import copy
+from strax import StorageFrontend, StorageBackend, Saver
 
 export, __all__ = strax.exporter()
 
 
 @export
 class MongoBackend(StorageBackend):
-
-    def __init__(self, uri, database):
+    """Mongo storage backend"""
+    def __init__(self, uri, database, col_n=None):
+        """
+        Backend for reading/writing data from Mongo
+        :param uri: Mongo url (with pw and username)
+        :param database: name of database (str)
+        :param col_n: collection name (str) to look for data
+        """
         if MongoClient is None:
             raise ImportError("Pymongo did not import: "
                               "cannot use MongoDBStore")
         self.client = MongoClient(uri)
         self.db = self.client[database]
+        self.col_name = col_n
 
     def _read_chunk(self, backend_key, chunk_info, dtype, compressor):
+        """See strax.Backend"""
+        query = backend_key_to_query(backend_key)
+        chunk_name = f'chunk_{chunk_info["chunk_i"]}'
+        # Query for the chunk and project the chunk info
+        doc = self.db[self.col_name].find_one(
+            {**query, chunk_name: {"$exists": True}},
+            {f"{chunk_name}.$": 1})
+        doc = doc.get(chunk_name, None)
```

Comment:
Might want to check first for `doc is None` in case the query doesn't find anything. (You know, the line beneath this one)

Reply:
Good catch, let me add it